### PR TITLE
C3 nohahesham s20 project delete

### PR DESCRIPTION
**Pull request by nohahesham**
@m0hamed 
@Fattouh92 
@SalmaFG 
@Omar-Hossam 
[issue #78]

### CODE REVIEW

File: arability/app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -22,9 +22,10 @@ class ProjectsController < BackendController
  #   project is not deleted
   def destroy
     @project = Project.find(params[:id])
+
     @project.destroy
     respond_to do |format|
-      format.html { redirect_to projects_url }
+      format.html { redirect_to :action => "index",:controller => "projects"}
```
Comment (omar-mohyeldin) [inline arability/app/controllers/projects_controller.rb]: avoid using :word1 => word2, use word1: word2 instead

File: arability/app/views/projects/index.html.erb [hunk 2]
```diff
@@ -52,7 +52,14 @@
   <% else %>
     <a class="btn pull-right" href="/developers/projects/<%= project1.id %>/edit"><i class="icon-edit"></i><%= t(:projects_index_edit)%></a>
   <% end %>
+
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb]: remove spaces from empty lines

**Comment by omar-mohyeldin** [comment]
@nohahesham please modify your code

**Comment by salmafg** [comment]
@nohahesham Documentation verified.

**Comment by abulazm22** [comment]
The UML is missing. please add it

**Comment by nohahesham** [comment]
@Omar-Hossam 
@m0hamed 
@Fattouh92 
plz review

**Comment by abulazm22** [comment]
Verified

File: arability/app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -5,11 +5,7 @@ class ProjectsController < BackendController
   # GET /projects.json
   before_filter :authenticate_gamer!
   before_filter :authenticate_developer!
-  before_filter :developer_can_see_this_project?, 
-  only: [:import_csv, :show, :add_from_csv_keywords, :choose_keywords]
-
-
-
+  before_filter :developer_can_see_this_project?, :only => [:import_csv, :show, :add_from_csv_keywords, :choose_keywords, :destroy]
  # author:Noha hesham
```
Comment (omar-mohyeldin) [inline arability/app/controllers/projects_controller.rb:11]: please add an empty line to differentiate between the "before filter" and the function in merging

File: arability/app/controllers/projects_controller.rb [hunk 4]
```diff
@@ -5,11 +5,7 @@ class ProjectsController < BackendController
   # GET /projects.json
   before_filter :authenticate_gamer!
   before_filter :authenticate_developer!
-  before_filter :developer_can_see_this_project?, 
-  only: [:import_csv, :show, :add_from_csv_keywords, :choose_keywords]
-
-
-
+  before_filter :developer_can_see_this_project?, :only => [:import_csv, :show, :add_from_csv_keywords, :choose_keywords, :destroy]
```
Comment (omar-mohyeldin) [inline arability/app/controllers/projects_controller.rb]: please user word1: [:word2] instead of :word  => [:word2]

File: arability/app/controllers/projects_controller.rb [hunk 5]
```diff
@@ -25,8 +21,9 @@ def destroy
     @project = Project.find(params[:id])
     @project.destroy
     respond_to do |format|
-      format.html { redirect_to projects_url }
+      format.html { redirect_to :action => "index",:controller => "projects"}
```
Comment (omar-mohyeldin) [inline arability/app/controllers/projects_controller.rb]: please use word1: "word2" instead of :word1 => "word2"

File: arability/app/models/developer.rb [hunk 6]
```diff
@@ -11,9 +11,7 @@ class Developer < ActiveRecord::Base
   has_many :projects_shared, :through => :shared_projects, :source => "project"
   has_many :projects, :foreign_key => "owner_id"
```
Comment (omar-mohyeldin) [inline arability/app/models/developer.rb]: remove empty line to make all the has_many together

File: arability/app/models/project.rb [hunk 7]
```diff
@@ -30,7 +29,7 @@ class Project < ActiveRecord::Base
 def self.createproject(params,developer_id)
   project = Project.new(params.except(:categories,:developer))
   project.owner_id = developer_id
-  project = createcategories(project,params[:categories])
+  # project = createcategories(project,params[:categories])
```
Comment (omar-mohyeldin) [inline arability/app/models/project.rb:32]: please remove commented lines from code

File: arability/app/views/developer/new.html.erb [hunk 8]
```diff
@@ -2,8 +2,7 @@
   <h3 class="form-signin-heading">
     <%= t(:register_as_developer) %>
   </h3>
-  <%= f.text_field :first_name, placeholder: t(:first_name), class: "input-block-level" %>
-  <%= f.text_field :last_name, placeholder: t(:last_name), class: "input-block-level" %>
+
```
Comment (omar-mohyeldin) [inline arability/app/views/developer/new.html.erb]: please remove empty lines inside html block of code if they are of no use, or at least remove the spaces if you need that empty line

File: arability/app/views/projects/index.html.erb (hunk 2 above)
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb]: you still didn't remove the spaces. You could also remove the whole empty line if it's of no use

File: arability/app/views/projects/index.html.erb [hunk 9]
```diff
@@ -52,7 +52,27 @@
   <% else %>
     <a class="btn pull-right" href="/developers/projects/<%= project1.id %>/edit"><i class="icon-edit"></i><%= t(:projects_index_edit)%></a>
   <% end %>
+  
+  <% if I18n.locale == :ar %>
+   <a style="margin-left : 6px;" href="#delete_modal" role="button" class="btn pull-left btn-danger" onClick="modalToggle()" id="LaunchDemo"><i class="icon-trash"></i><%= t(:projects_index_delete)%></a>
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb]: line should be indented by 2 spaces not only 1

File: arability/app/views/projects/index.html.erb [hunk 10]
```diff
@@ -52,7 +52,27 @@
...
+  <% else %>
+  <a style="margin-right : 6px;" href="#delete_modal" role="button" class="btn pull-right btn-danger" onClick="modalToggle()" id="LaunchDemo"> <i class="icon-trash"></i><%= t(:projects_index_delete)%></a>
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb]: please indent line with 2 spaces

File: arability/app/views/projects/index.html.erb [hunk 11]
```diff
@@ -52,7 +52,27 @@
...
+  <% end %>
+    <div id="delete_modal" class="modal hide fade" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb:60]: please allign this line with the one above

File: arability/app/views/projects/index.html.erb [hunk 12]
```diff
@@ -52,7 +52,27 @@
...
+    <div class="modal-header">
+      <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+      <h3 id="modal_label"><%= t(:delete) %></h3>
+      </div>
+      <div class="modal-body">
+        <h4><%= t(:confirm)%></h4>
+      </div>
+      <div class="modal-footer">
+        <button class="btn" data-dismiss="modal"><%= t(:close_delete) %></button>
+        <%= link_to t(:submit_delete), delete_path(project1.id),:class => "btn btn-danger" %>
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb]: please leave space after comma, and use word1: "word2" instead of :word1 => "word2"

File: arability/app/views/projects/index.html.erb [hunk 13]
```diff
@@ -52,7 +52,27 @@
...
+  <% if I18n.locale == :ar %>
+   <a style="margin-left : 6px;" href="#delete_modal" role="button" class="btn pull-left btn-danger" onClick="modalToggle()" id="LaunchDemo"><i class="icon-trash"></i><%= t(:projects_index_delete)%></a>
+  <% else %>
+  <a style="margin-right : 6px;" href="#delete_modal" role="button" class="btn pull-right btn-danger" onClick="modalToggle()" id="LaunchDemo"> <i class="icon-trash"></i><%= t(:projects_index_delete)%></a>
+  <% end %>
+    <div id="delete_modal" class="modal hide fade" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
+    <div class="modal-header">
+      <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+      <h3 id="modal_label"><%= t(:delete) %></h3>
+      </div>
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb]: please allign the opening div with the closing div

File: arability/app/views/projects/index.html.erb [hunk 14]
```diff
@@ -52,7 +52,27 @@
...
+      <div class="modal-body">
+        <h4><%= t(:confirm)%></h4>
+      </div>
+      <div class="modal-footer">
+        <button class="btn" data-dismiss="modal"><%= t(:close_delete) %></button>
+        <%= link_to t(:submit_delete), delete_path(project1.id),:class => "btn btn-danger" %>
+      </div>
+    </div>
   <br />
+
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb]: please re-allign all the above lines correctly after alligning the lines commented above, and remove empty line IF it's of no use

File: arability/app/views/projects/index.html.erb [hunk 15]
```diff
@@ -86,9 +106,27 @@
   <br />
   <% if I18n.locale == :ar %>
     <a class="btn pull-left" href="/developers/projects/<%= project2.id %>/edit"><i class="icon-edit"></i><%= t(:projects_index_edit)%></a>
-  <% else %>
     <a class="btn pull-right" href="/developers/projects/<%= project2.id %>/edit"><i class="icon-edit"></i><%= t(:projects_index_edit)%></a>
   <% end %>
+
+   <% if I18n.locale == :ar %>
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb:111]: please re-allign this line, and all the coming lines correctly accordingly

File: arability/app/views/projects/index.html.erb [hunk 16]
```diff
@@ -86,9 +106,27 @@
...
+    <div class="modal-header">
+      <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+      <h3 id="modal_label"><%= t(:delete) %></h3>
+      </div>
+      <div class="modal-body">
+        <h4><%= t(:confirm)%></h4>
+      </div>
+      <div class="modal-footer">
+        <button class="btn" data-dismiss="modal"><%= t(:close_delete) %></button>
+        <%= link_to t(:submit_delete), delete_path(project1.id),:class => "btn btn-danger" %>
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb]: please leave space after comma, and use word1: "word2" instead of :word1 => "word2"

File: arability/app/views/projects/index.html.erb [hunk 17]
```diff
@@ -193,4 +231,8 @@
       </p>
     </div>
   </div>
-</div>
\ No newline at end of file
+</div>
+
+<script type="text/javascript">
+<%= render "delete.js" %>
```
Comment (omar-mohyeldin) [inline arability/app/views/projects/index.html.erb]: indent line

File: arability/config/initializers/z_bootstrap.rb [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+Keyword.create(:name => "find", :approved => true, :is_english => true)
+Keyword.create(:name => "search", :approved => true, :is_english => true)
+Keyword.create(:name => "hide", :approved => true, :is_english => true)
+Keyword.create(:name => "أنشر", :approved => true, :is_english => false)
+Keyword.create(:name => "دوس", :approved => true, :is_english => false)
+Keyword.create(:name => "يعجبني", :approved => true, :is_english => false)
+Keyword.create(:name => "عربية", :approved => true, :is_english => false)
+Keyword.create(:name => "مرحلة", :approved => true, :is_english => false)
+Keyword.create(:name => "إعدادات", :approved => true, :is_english => false)
+Keyword.create(:name => "انجازات", :approved => true, :is_english => false)
```
Comment (omar-mohyeldin) [inline arability/config/initializers/z_bootstrap.rb:25]: please follow the code convention of using word1: word2 instead of :word1 => word2

File: arability/config/routes.rb [hunk 19]
```diff
@@ -66,10 +65,16 @@
       match '/projects/:project_id/export_csv' => "projects#export_to_csv", :as => "projects_export_csv"
       match '/projects/:id/import_csv' => "projects#import_csv", :as => :import_csv_project
       match '/projects/:id/choose_keywords' => "projects#choose_keywords", :as => :choose_keywords_project
+
+      match '/projects/:id/destroy' => "projects#destroy", :as => :delete
```
Comment (omar-mohyeldin) [inline arability/config/routes.rb]: please use double quotations instead of single quotations

File: arability/config/routes.rb [hunk 20]
```diff
@@ -66,10 +65,16 @@
...
+      put "projects/destroy"
+
       match '/projects/:project_id/export_xml' => "projects#export_to_xml", :as => "projects_export_xml"
       match '/projects/:project_id/export_json' => "projects#export_to_json", :as => "projects_export_json"
+
       resources :projects
 
+
```
Comment (omar-mohyeldin) [inline arability/config/routes.rb]: why is there 2 empty lines? please remove any unneeded empty lines you added in file

File: arability/spec/controllers/projects_controller_spec.rb [hunk 21]
```diff
@@ -57,6 +57,12 @@
     page.should have_content(I18n.t(:import_csv_title))
   end
 
+    it "should delete a project" do
+    sign_in gamer1
+    put :destroy, :id => project.id
```
Comment (omar-mohyeldin) [inline arability/spec/controllers/projects_controller_spec.rb]: follow code conventions, don't use =>

File: arability/spec/controllers/projects_controller_spec.rb [hunk 22]
```diff
@@ -57,6 +57,12 @@
...
+    #response.should be_success
```
Comment (omar-mohyeldin) [inline arability/spec/controllers/projects_controller_spec.rb]: remove commented lines

File: arability/spec/controllers/projects_controller_spec.rb [hunk 23]
```diff
@@ -100,5 +106,4 @@
     get :export_to_json, project_id: p.id
     response.code.should eq("200")
   end
-
```
Comment (omar-mohyeldin) [inline arability/spec/controllers/projects_controller_spec.rb:103]: why did you remove empty line? it should be kept for git merging conflicts

**Comment by omar-mohyeldin** [comment]
@nohahesham modify your code

File: arability/app/models/project.rb (hunk 7 above)
Comment (nohahesham) [inline arability/app/models/project.rb:32]: its not my code so i dont wanna remove it , but it makes an error cause column categories is removed

File: arability/spec/controllers/projects_controller_spec.rb [hunk 24]
```diff
@@ -57,6 +57,12 @@
     page.should have_content(I18n.t(:import_csv_title))
   end
 
+    it "should delete a project" do
```
Comment (salmafg) [inline arability/spec/controllers/projects_controller_spec.rb]: Can't mass-assign protected attributes: name_en, name_ar (ActiveModel::MassAssignmentSecurity::Error)

File: arability/config/initializers/z_bootstrap.rb (hunk 18 above)
Comment (nohahesham) [inline arability/config/initializers/z_bootstrap.rb:25]: not my code at all

**Comment by nohahesham** [comment]
@SalmaFG 
@Omar-Hossam

**Comment by salmafg** [comment]
Testing verified.

File: arability/spec/controllers/projects_controller_spec.rb [hunk 25]
```diff
@@ -100,5 +104,5 @@
     get :export_to_json, project_id: p.id
     response.code.should eq("200")
   end
-
+
```
Comment (omar-mohyeldin) [inline arability/spec/controllers/projects_controller_spec.rb]: remove spaces from empty line

**Comment by omar-mohyeldin** [comment]
@nohahesham code verified
